src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -869,41 +762,42 @@ private static string GetLocalizedNameByMuiNativeResource(string resource)
                 return string.Empty;
             }
 
-            string filePath;
+            // Get the resource ID
+            if (!int.TryParse(resources[1], NumberStyles.Integer, CultureInfo.InvariantCulture, out int resourceId))
+            {
+                return string.Empty;
+            }
+            resourceId = -resourceId;
+
+            // Use the current UI culture when culture not specified
+            CultureInfo cultureInfo = CultureInfo.CurrentUICulture;
 
             // get the path to Windows\System32
             string system32 = Environment.SystemDirectory;
 
             // trim the string "@tzres.dll" => "tzres.dll"
```

Comment:
Does this comment need to be updated as well?

Reply:
This commend should have been removed in #54247.